sdk/loadtestservice/azure-developer-loadtesting/src/main/java/com/azure/developer/loadtesting/LoadTestAdministrationAsyncClient.java

```diff
@@ -38,314 +43,322 @@ public final class LoadTestAdministrationAsyncClient {
      *             id: String (Optional)
      *             resourceId: String (Required)
-     *             metricnamespace: String (Required)
+     *             metricNamespace: String (Required)
      *             displayDescription: String (Optional)
-     *             name (Required): {
-     *                 value: String (Required)
-     *                 localizedValue: String (Required)
-     *             }
+     *             name: String (Required)
      *             aggregation: String (Required)
      *             unit: String (Optional)
      *             resourceType: String (Required)
      *         }
      *     }
+     *     createdDateTime: OffsetDateTime (Optional)
+     *     createdBy: String (Optional)
+     *     lastModifiedDateTime: OffsetDateTime (Optional)
+     *     lastModifiedBy: String (Optional)
      * }
      * }</pre>
      *
-     * @param name Unique name for server metrics, must be a valid URL character ^[a-z0-9_-]*$.
+     * @param testId Unique name for the load test, must contain only lower-case alphabetic, numeric, underscore or
+     *     hyphen characters.
      * @param requestOptions The options to configure the HTTP request before HTTP client sends it.
      * @throws HttpResponseException thrown if the request is rejected by server.
      * @throws ClientAuthenticationException thrown if the request is rejected by server on status code 401.
      * @throws ResourceNotFoundException thrown if the request is rejected by server on status code 404.
      * @throws ResourceModifiedException thrown if the request is rejected by server on status code 409.
-     * @return server metrics configuration by its name along with {@link Response} on successful completion of {@link
-     *     Mono}.
+     * @return server metric configuration for the given test along with {@link Response} on successful completion of
+     *     {@link Mono}.
      */
     @Generated
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<Response<BinaryData>> getServerMetricsByNameWithResponse(String name, RequestOptions requestOptions) {
-        return this.serviceClient.getServerMetricsByNameWithResponseAsync(name, requestOptions);
+    public Mono<Response<BinaryData>> getServerMetricsConfigWithResponse(String testId, RequestOptions requestOptions) {
+        return this.serviceClient.getServerMetricsConfigWithResponseAsync(testId, requestOptions);
+    }
+
+    /** Validation status of uploaded file */
+    public enum ValidationStatus {
+
+        /** Validation not initiated */
+        NOT_VALIDATED,
+        /** Validation successful */
+        VALIDATION_SUCCESS,
+        /** Validation failed */
+        VALIDATION_FAILURE,
+        /** Validation initiated */
+        VALIDATION_INITIATED,
+        /** Validation not required for file type */
+        VALIDATION_NOT_REQUIRED
+    }
+
+    private Mono<PollResponse<ValidationStatus>> getValidationStatus(BinaryData fileMono) {
```

Comment:
I am inclined to return `PollResponse<ValidationStatus>`, as nothing async is done in this method.

Reactor should convert a throwed exception in its `map` operator to Mono.error automatically. (though worth a quick test, e.g. modify the code to just throw one and see result)

Reply:
Throwing exceptions is not allowed it seems. However, wrapping it around Mono.error is allowed. Is there any workaround for this, as Mono<PollResponse<>> is changed to PollResponse<>?

`Directly throwing an exception is disallowed. Must throw through "ClientLogger" API, either of "logger.logExceptionAsError", "logger.logThrowableAsError", "logger.atError().log", "logger.logExceptionAsWarning", "logger.logThrowableAsWarning", or "logger.atWarning().log" where "logger" is type of ClientLogger from Azure Core package. [ThrowFromClientLogger]`

Reply:
Yes, exception need to be thrown via ClientLogger API. It is the standard procedure https://azure.github.io/azure-sdk/java_implementation.html#java-logging-log-and-throw (my take is that if user can see this exception, it should be thrown via ClientLogger; you may try do the Mono.error and see if user will get it in sync API).

Sure, you can leave it be, for now. As it is pretty low-impact.